code/datums/components/laser_components.dm

```diff
@@ -0,0 +1,59 @@
+///it handles adding and removing the projectile
+/datum/component/extralasers
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	///the path of the external lens object, used it when it has to be spawned
+	var/lens_path
+	///the laser object
+	var/obj/item/ammo_casing/energy/ammo
+
+/datum/component/extralasers/Initialize(_ammo, _lens_path)
+	if(!istype(parent, /obj/item/gun/energy/laser))
+		return COMPONENT_INCOMPATIBLE
+	ammo = _ammo
+	lens_path = _lens_path
+
+/datum/component/extralasers/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/attackby)
+	attach()
+
+/datum/component/extralasers/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_PARENT_ATTACKBY)
+	detach()
+
+/datum/component/extralasers/Destroy()
+	detach()
+	qdel(ammo)
+	return ..()
+
+///it new() the projectile and adds it in the ammo types
+/datum/component/extralasers/proc/attach()
+	var/obj/item/gun/energy/laser/L = parent
+	ammo =  new ammo (src)
+	L.ammo_type  += ammo
+
+///removes the projectile from available ammo types
+/datum/component/extralasers/proc/detach()
+	var/obj/item/gun/energy/laser/L = parent
+	if(L.chambered)
+		L.chambered = null
+	LAZYREMOVE(L.ammo_type, ammo)
+	L.select_fire()
+	L.recharge_newshot()
+	L.update_icon(TRUE)
+
+/datum/component/extralasers/proc/attackby(datum/source, obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_CROWBAR)
```

Comment:
This is still really ugly, you should make a crowbar act signal. See
https://github.com/tgstation/tgstation/blob/28f0a5a890d570df99a02ff94767d4208ca35917/code/game/atoms.dm#L937-L939

Copy that code into the crowbar_act above, then a new define under here, copying everything except replacing screwdriver with crowbar:
https://github.com/tgstation/tgstation/blob/28f0a5a890d570df99a02ff94767d4208ca35917/code/__DEFINES/components.dm#L82